### PR TITLE
DOC: Improved docs for jnp.ptp and count_nonzero

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
This PR updates the docstrings of `jnp.ptp` and `jnp.count_nonzero`.

Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -1069,9 +1069,44 @@ def _std(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
   return lax.sqrt(var(a, axis=axis, dtype=dtype, correction=correction, keepdims=keepdims, where=where))
 
 
-@implements(np.ptp, skip_params=['out'])
 def ptp(a: ArrayLike, axis: Axis = None, out: None = None,
         keepdims: bool = False) -> Array:
+  r"""Return the range of values (maximum - minimum) along a given axis.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1074]: I think this is a bit too similar to the numpy verbiage. How about something like the "peak-to-peak range", or explicitly the "difference between the maximum and minimum values"?

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -1069,9 +1069,44 @@ def _std(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
...
+
+  JAX implementation of :func:`numpy.ptp`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      range is computed. If None, the range is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1081]:
```suggestion
      range is computed. If None, the range is computed on the flattened array.
```

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -1085,11 +1120,45 @@ def _ptp(a: ArrayLike, axis: Axis = None, out: None = None,
...
                   keepdims: bool = False) -> Array:
   check_arraylike("count_nonzero", a)
+  r"""Return the number of nonzero elements along a given axis.
+
+  JAX implementation of :func:`numpy.count_nonzero`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      number of nonzeros are counted. If None, counts along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:1134]:
```suggestion
      number of nonzeros are counted. If None, counts within the flattened array.
```

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks - looks good! Some small changes requested.

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Changed

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Hi @dfm, implemented the changes suggested and squashed into single commit.

**Comment by dfm** [comment]
Looks good now - thanks! Can you squash?

**Comment by rajasekharporeddy** [comment]
Squashed into single commit.

**Review by dfm — APPROVED** [review]

**Pull request merged**